BRANCHING_RELEASE_STRATEGY.md

```diff
@@ -118,16 +142,28 @@ You can then push this branch.
 ```shell
 $ git checkout -b tmp_prepare_release
 $ mvn versions:set -DnewVersion=X.Y.0
-$ git commit -s -m "Bump to vX.Y.0"
+$ git commit -s -a -S -m "Bump to vX.Y.0"
```

Comment:
Should these two options (-a -S) also be used in the commit commands or only the last commit before the tag has to be signed?

Reply:
- `-a` is for adding all the modified files to the index (so that they will be taken into account by the `commit` command. 

Here the modified files are the `pom.xml` that have been affected by the previous `mvn versions:set ...` command.

In the previous documentation version, it was missing: if you simply copied and pasted the given commands without much thinking, the modified files were not added to your commit. An empty commit is not taken into account by git so it was not problematic but rather annoying.

An argument againt this `-a` option is that you can unvoluntarily add other files to your index. Here, you are not supposed to have other modified files however. 
Maybe we should add a `git status` step before the `git commit -s -a -S -m "Bumpt to vX.Y.0"` so that the developper can make sure that the only modified files are the `pom.xml`?

- `-S` is for signing **the commit that is used to create the tag** with the PGP/GPG key. 
(It is not necessarily the last commit. Here, for example, we have another commit that is "Bump to VX.Y+1.0-SNAPSHOT".)

The commit used to create the tag, as well as the tag should be signed with the PGP/GPG key.

Reply:
I was thinking of the commit "Bump to VX.Y+1.0-SNAPSHOT" so if it's not necessary to sign it, it looks good to me.

Reply:
Yeah, this one does not need to be signed with the PGP/GPG key.